### PR TITLE
Deterministic flag in options wich initializes the random generator w…

### PR DESCRIPTION
**Pull request by a-mueller**
I added a deterministic flag to the options. If set the word cloud will always look the same given the same input.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -101,6 +102,14 @@ function Wordcloud({
         .sort((x, y) => descending(x.value, y.value))
         .slice(0, maxWords);
 
+      let randomGenerator = Math.random;
+
+      if (options.deterministic){
+        const seedrandom = require('seedrandom');
```
Comment (chrisrzhou) [inline src/index.tsx]: I'm okay with moving this to the top and incurring a slight hit.  You can use ES6 `import` syntax.

Once this is moved to the top, you can use `seedrandom` to initialize `randomGenerator`, so something like

```
const randomGenerator = options.deterministic ? seedrandom() : seedrandom('deterministic');
```

File: src/types.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ export interface Callbacks {
 export interface Options {
   colors: string[];
   enableTooltip: boolean;
+  deterministic: boolean;
```
Comment (chrisrzhou) [inline src/types.ts:29]: small nit: put above row 29 (alphabetical)

**Review by chrisrzhou — COMMENTED** [review]
This is a great suggestion and thank you for including it!

If you are adventurous, feel free to update the `docz` documentation for the `Options` MDX file.  Feel free to add a short section about using the `deterministic` option field. You can build the docs and test them locally with `yarn dev`.

I'll be making slight improvement to the typescript types this weekend, and reducing bundle size, and I can include your changes for a next version release soon!

**Comment by a-mueller** [comment]
Hey,
I made some of the changes you suggested. My typescript knowledge unfortunately was not enough to make the import work instead of using require. Kept having issues with default exports / namespace not being callable. I did extend the commit to include the rotate and colour choosing as well.

File: src/index.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import render from './render';
 import { Callbacks, MinMaxPair, Options, Scale, Spiral, Word } from './types';
 import { getDefaultColors, getFontScale, getText, rotate } from './utils';
 
+
```
Comment (chrisrzhou) [inline src/index.tsx]: remove extra line

File: src/index.tsx [hunk 4]
```diff
@@ -101,6 +103,12 @@ function Wordcloud({
         .sort((x, y) => descending(x.value, y.value))
         .slice(0, maxWords);
 
+
+      const pmr = require('pseudo-math-random');
```
Comment (chrisrzhou) [inline src/index.tsx]: Move to top of file (maybe after line 8)

File: src/index.tsx [hunk 5]
```diff
@@ -101,6 +103,12 @@ function Wordcloud({
...
+      const randomGenerator: ()=> number = options.deterministic ? pmr('deterministic') : pmr();
```
Comment (chrisrzhou) [inline src/index.tsx]: You can fix a few code formatting issues by running `yarn lint`.  It should fix up most of the formatting with the prettier rules in the codebase.

Comment (chrisrzhou) [reply src/index.tsx]: You can probably call this `random`, since it effectively is a callback replacing `Math.random` in the code here.

**Review by chrisrzhou — APPROVED** [review]
Hey thanks a lot for updating the docs and generalizing `Math.random` calls with `pmr`.  This looks good to me.  Please address a few things before merging in

- rename `randomGenerator` with `random`
- Run `yarn lint` after code changes so `prettier` can auto format the codebase.
- With regards to the `yarn.lock` file and error you're seeing, I'm not sure if deleting the lock file and running `yarn` can help solve the problem.  I can help look into the issue in a future PR.

Thanks!

**Comment by a-mueller** [comment]
Done.

Had to "eslint-disable-line" the pmr import after moving it to the suggested position. Didn't want to take the rule out the lint config alltogether but feel free to change that.

**Review by chrisrzhou — APPROVED** [review]

**Comment by chrisrzhou** [comment]
Thanks so much for the changes here and enhancing the component!

**Pull request merged**